firewood/src/merkle.rs

```diff
@@ -408,18 +412,17 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
                             u.rehash();
                         })
                         .unwrap();
+
                         break;
                     }
                 },
+
                 NodeType::Leaf(n) => {
                     // we collided with another key; make a copy
                     // of the stored key to pass into split
                     let n_path = n.0.to_vec();
                     let n_value = Some(n.1.clone());
-                    let rem_path = key_nibbles
-                        .into_iter()
-                        .skip(key_nib_offset)
-                        .collect::<Vec<_>>();
+                    let rem_path = once(key_nib).chain(key_nibbles).collect::<Vec<_>>();
```

Comment:
nit: very nice, but took some digesting; maybe rename `key_nib` -> `current_nibble`

Reply:
Will also do in a follow-up. I didn't love it either, but I wanted to do away with offsets to make the overall function more digestible.